[FE-112] Boundary Enterprise a11y audit- User details form

venusang opened the pull request:
# Description
This updates the input fields on the User details page so that they are `readonly` instead of `disabled`.  Additionally it changes the edit button text from `Edit form` to `Edit User details` so that it is more descriptive.  Lastly, `aria-describedby` was also added to the button so that it is a11y compliant.  

<!-- Uncomment line below to manually add link to JIRA ticket -->
:tickets: [FE-112](https://hashicorp.atlassian.net/browse/FE-112)

## Screenshots (if appropriate)

<img width="3840" height="1924" alt="fe-112-before-after" src="https://github.com/user-attachments/assets/46c855eb-6c66-4d8b-9885-dcae69ed3f4d" />

## How to Test
Navigate to a User in the Users list.  Verify that the input fields are readonly (you should be able to highlight the text with your mouse) and that the submit button reflects the text changes shown in the after screenshot.

## Checklist
<!-- strikethrough the checklist item that is not relevant to your change -->

<!-- If you are merging a long lived branch, major feature, or UI altering changes,
please add the a11y-tests label. Other cases, like a dependency change or
translation update likely does not need an a11y audit -->

- [x] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [x] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added `a11y-tests` label to run a11y audit tests if needed

## PCI review checklist

<!-- heimdall_github_prtemplate:grc-pci_dss-2024-01-05 -->
- [x] I have documented a clear reason for, and description of, the change I am making.
- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.
- [ ] If applicable, I've documented the impact of any changes to security controls.
  Examples of changes to security controls include using new access control methods, adding or removing logging pipelines, etc.


[FE-112]: https://hashicorp.atlassian.net/browse/FE-112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

venusang committed:
refactor: 💡 add edit-user-details translations

This adds edit user details translations so the User detail form submit
button is a11y compliant

✅ Closes: FE-112

venusang committed:
refactor: 💡 change inputs to readonly and update submit button

Per a11y audit, this changes the input fields to readonly instead of
disabled. It also updates the submit button text so that it is context
based and has an aria-label for screenreaders.

✅ Closes: FE-112

venusang committed:
refactor: 💡 move translation to resources for user

This moves the User detail action translation to resources/en-us.yaml
since it is specific to the user resource and is not a generic form
action.

✅ Closes: FE-112

cameronperera reviewed (dismissed):
LGTM

venusang commented on ui/admin/app/components/form/user/index.hbs:
```diff
@@ -52,7 +52,8 @@
   {{#if (can 'save model' @model)}}
     <form.actions
-      @enableEditText={{t 'actions.edit-form'}}
+      @enableEditText={{t 'resources.user.actions.edit-details.button.text'}}
+      aria-label={{t 'resources.user.actions.edit-details.button.aria-label'}}
```
I implemented this based on the recommendations outlined in the ticket [here](https://hashicorp.atlassian.net/browse/FE-112).

> 8.2 “Edit Form” button text
> Issue: Due to the structure of the page content, it was really difficult to tell what form I would be editing while using only a screen reader to navigate the page. I would consider adding the form name to the button so that it would be more informative for the user. Suggest adding an aria-label attribute to the button, where the value is “edit form: users”. Note that the visible button label matches the first part of the aria-label attribute value, conforming to [WCAG Success Criterion 2.5.3 Label In Name](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html).

ZedLi replied:
I took the ticket to mean to add an `aria-label` to make the existing `Edit form` button more descriptive for screen readers such that `Edit form: users` now properly describes the existing button. I also assume just changing the text of the button without an aria label would have accomplished that but it sounds like the recommendation is to add the label. 

But either way reading the linked WCAG success criteria it seems like this change would now violate it:
> Accessible name matches visible label: The accessible name and visible label of a control match.
> Accessible name starts with visible label: The accessible name "Search for a value" begins with the text of the visible label, "Search".

Since the text is now `Edit user details` and the label is `Edit form: users`, they don't match and the accessible name doesn't start with the visible label (as opposed to `Edit form: users` just being a more descriptive match of the original text).

venusang committed:
refactor: 💡 removed aria-label from button set

This was not sending the aria-label down to the Edit Users detail button.
However, taking a look further at Hds::Button, this component
does not allow aria-label to be set unless it is an icon only button.
With that said, this cleans up the translation structure and removes the
aria-label translation and mark-up.

✅ Closes: FE-112

ZedLi reviewed (dismissed):
Looks good! Thanks for the update

hashicc reviewed (dismissed):
Thanks for helping us with our a11y!

venusang committed:
fix: 🐛 Add aria-describedby to Edit user details button

Since we cannot apply aria-label to an Hds::Button
when it has text, we can alternatively apply the
aria-describedby attribute instead.  Its value
must match the id of the form.  With that said, a
form id value was also added.

✅ Closes: FE-112

ZedLi commented on ui/admin/app/components/form/user/index.hbs:
```diff
@@ -52,7 +53,10 @@
   {{#if (can 'save model' @model)}}
     <form.actions
-      @enableEditText={{t 'actions.edit-form'}}
+      @ariaDescribedBy={{t
```
I'm not sure I understand why we linked the ID of the form itself, what text are we trying to descriptively add to this by using `aria-describedby`? It also seems weird to use a translated text as the ID when the element itself should have the translated text

venusang replied:
I pinged you in the JIRA ticket comment but adding reference here for good measure.

(taking a screenshot so it has more context versus copy paste):

<img width="1650" height="1092" alt="CleanShot 2025-09-10 at 09 25 10@2x" src="https://github.com/user-attachments/assets/f3e319ed-6f11-4c8e-8dd4-4167d76d140e" />

https://hashicorp.atlassian.net/browse/FE-112?focusedCommentId=828913

venusang replied:
Going off @MelSumner thoughts in the JIRA ticket, I set the form ID value to be `user-details` versus `users` since to me it seems to make better sense that the form is specific to the `user-details`.  `Users` did not seem to make sense to me but maybe I am misunderstanding the purpose here.  

> It also seems weird to use a translated text as the ID when the element itself should have the translated text

@ZedLi from what I am getting here, it sounds like the form ID value should be semantic versus the ember generated ID.  Maybe Melanie could provide some clarity here for both of us.  I'm not all that familiar with `aria-describedby` prior to working on this.

ZedLi replied:
Right, I think all those choices make sense but it sounds like we just need to pick one choice? It seems we could leave the button as `Edit form` if we're adding one of the aria attributes. 

With regards to the current implementation with 3, I was specifically pointing out the fact that it's currently pointing to the form ID instead of something with descriptive text (which melanie suggests as the form title):

> where the value is the ID of the form title element (in this case, the element with the text “Users”)

Though that looks like it has some extra text which would pulled in as part of the helper doc link text.

nikkipurcell commented on addons/core/translations/resources/en-us.yaml:
```diff
@@ -570,6 +570,9 @@ user:
     delete: Delete User
+    edit-details:
+      button-text: 'Edit User details'
+      aria-describedby: 'user-details'
```
change aria-describedby to 'edit-user-details' to match button text

venusang committed:
refactor: 💡 update translation and removed unnecessary aria

This updates the translation to be more explicit to edit user details
and removes the unnecessary aria at attribute in the form

✅ Closes: FE-112

ZedLi commented on addons/core/translations/resources/en-us.yaml:
```diff
@@ -571,6 +571,9 @@ user:
     delete: Delete User
+    edit-details:
+      button-text: 'Edit User details'
+      aria-describedby: 'edit-user-details'
```
I don't think we should have a translation for an `aria-describedby`
